src/weightedcoorddesc.jl

```diff
@@ -0,0 +1,83 @@
+mutable struct WeightedCoordinateDescent{T}
+    M::Matrix{T}
+    maxiter::Int           # maximum number of iterations (in main procedure)
+    verbose::Bool          # whether to show procedural information
+    tol::T                 # tolerance of changes on W and H upon convergence
+    update_H::Bool         # whether to update H
+    shuffle::Bool          # # if true, randomize the order of coordinates in the CD solver           # mask matrix
+    function WeightedCoordinateDescent{T}(M::Matrix{T};
```

Comment:
```suggestion
    function WeightedCoordinateDescent{T}(M::AbstractMatrix;
```

If you call `WeightedCoordinateDescent{Float64}([1 0; 0 1])` (with a `Matrix{Int}` argument), your version will throw a `MethodError`, because it forces the user to convert `M` to `Matrix{Float64}` before calling the constructor. My version will be accepted and it will convert `M` to a `Matrix{Float64}`.